include/zephyr/drivers/charger.h

```diff
@@ -99,6 +183,18 @@ union charger_propval {
 	bool present;
 	/** CHARGER_PROP_STATUS */
 	enum charger_status status;
+	/** CHARGER_PROP_CHARGE_TYPE */
+	enum charger_charge_type charge_type;
+	/** CHARGER_PROP_HEALTH */
+	enum charger_health health;
+	/** CHARGER_PROP_CONSTANT_CHARGE_CURRENT */
+	uint32_t const_charge_curr_ua;
+	/** CHARGER_PROP_PRECHARGE_CURRENT */
+	uint32_t precharge_curr_ua;
+	/** CHARGER_PROP_CHARGE_TERM_CURRENT */
+	uint32_t charge_term_curr_ua;
+	/** CHARGER_PROP_CONSTANT_CHARGE_VOLTAGE */
+	uint32_t const_charge_volt_uv;
```

Comment:
Ok but now the fields and property names are incoherent, for example the prop has "current" but the field is "curr_ua". I think that the two should match beside the prefix and case, what do you think?

So you'd have something like

```c

	case CHARGER_PROP_PRECHARGE_CURRENT_UA:
		return bq24190_charger_get_pc(dev, &val->precharge_current_ua);
```

and there would be no ambiguity between matching the property and the union field.